File: Android/TagMaster/src/depollsoft/tagmaster/SettingsModel.java
package depollsoft.tagmaster;

import depollsoft.lib.activity.RichApplication;
import depollsoft.lib.util.ContentCache;
import depollsoft.lib.util.Preferences;

public class SettingsModel {
  private static final String minimumRandomTagRatingKey = "tagmaster.MinimumRandomTagRating";
  private static final String minimumRandomDownloadsKey = "tagmaster.MinimumRandomDownloadsKey";
  private static final String sheetMusicRandomKey = "tagmaster.SheetMusicRandomKey";
  private static final String learningTracksRandomKey = "tagmaster.LearningTracksRandomKey";

  static {
    Preferences.initialize(SettingsModel.minimumRandomDownloadsKey, 100,
        Integer.class);
    Preferences.initialize(SettingsModel.minimumRandomTagRatingKey, 2.5d,
        Double.class);
    Preferences.initialize(SettingsModel.sheetMusicRandomKey, true,
        Boolean.class);
    Preferences.initialize(SettingsModel.learningTracksRandomKey, null,
        Boolean.class);
  }

  public static void clearCache() {
    ContentCache cc = new ContentCache(RichApplication.getAppContext());
    cc.clearCache();
  }

  public static double getCacheSizeInMegabytes() {
    ContentCache cc = new ContentCache(RichApplication.getAppContext());
    return cc.getCacheSize() / 1024d / 1024d;
  }

  public static int getMinimumRandomDownloads() {
    return Preferences.<Integer>get(SettingsModel.minimumRandomDownloadsKey);
  }

  public static double getMinimumRandomTagRating() {
    return Preferences.<Double>get(SettingsModel.minimumRandomTagRatingKey);
  }

  public static Boolean getRandomLearningTracksFilter() {
    return Preferences.get(SettingsModel.learningTracksRandomKey);
  }

  public static Boolean getRandomSheetMusicFilter() {
    return Preferences.get(SettingsModel.sheetMusicRandomKey);
  }

  public static void setMinimumRandomDownloads(int value) {
    Preferences.set(SettingsModel.minimumRandomDownloadsKey, value);
  }

  public static void setMinimumRandomTagRating(double value) {
    Preferences.set(SettingsModel.minimumRandomTagRatingKey, value);
  }

  public static void setRandomLearningTracksFilter(Boolean value) {
    Preferences.set(SettingsModel.learningTracksRandomKey, value);
  }

  public static void setRandomSheetMusicFilter(Boolean value) {
    Preferences.set(SettingsModel.sheetMusicRandomKey, value);
  }

}

File: Android/TagMaster/src/main/java/depollsoft/tagmaster/SettingsModel.kt
package depollsoft.tagmaster

import depollsoft.lib.activity.RichApplication
import depollsoft.lib.util.ContentCache
import depollsoft.lib.util.preference

object SettingsModel {
    private const val minimumRandomTagRatingKey = "tagmaster.MinimumRandomTagRating"
    private const val minimumRandomDownloadsKey = "tagmaster.MinimumRandomDownloadsKey"
    private const val sheetMusicRandomKey = "tagmaster.SheetMusicRandomKey"
    private const val learningTracksRandomKey = "tagmaster.LearningTracksRandomKey"
    private const val wakeLockKey = "tagmaster.WakeLockOnSheetMusic"
    fun clearCache() {
        val cc = ContentCache(RichApplication.getAppContext())
        cc.clearCache()
    }

    val cacheSizeInMegabytes: Double
        get() {
            val cc = ContentCache(RichApplication.getAppContext())
            return cc.cacheSize / 1024.0 / 1024.0
        }
    var minimumRandomDownloads: Int by preference(minimumRandomDownloadsKey, 100)
    var minimumRandomTagRating: Double by preference(minimumRandomTagRatingKey, 2.5)
    var randomLearningTracksFilter: Boolean? by preference(learningTracksRandomKey, null)
    var randomSheetMusicFilter: Boolean? by preference(sheetMusicRandomKey, true)
    var wakeLockOnSheetMusic: Boolean by preference(wakeLockKey, true)
}
